src/Controls/src/Core/Menu/MenuFlyoutSubItem.cs

```diff
@@ -16,7 +16,12 @@ public partial class MenuFlyoutSubItem : MenuFlyoutItem, IMenuFlyoutSubItem
 
 		public MenuFlyoutSubItem()
 		{
-			LogicalChildrenInternalBackingStore = new CastingList<Element, IMenuElement>(_menus);
```

Comment:
I think the casting list is a wrapper and if you change the source is is also reflected.

Adding new items to a new list is not going to keep them synced.

Can we use the casting list still and just go to Element?

Reply:
You are right, applied changes to fix it.